brightcove/brightcove/src/page/Home.tsx

```diff
@@ -0,0 +1,204 @@
+        const data = {
+          authUrl: extensions.config.oauthUrl,
+          videoUrl: `${extensions.config.brightcoveUrl}/${ids}`,
+        };
+        const brightcove = new Brightcove(extensions.config.proxyUrl);
+        brightcove.getVideos(data).then(({ data }) => {
+          setVideoList(data);
+        });
+      } else {
+        setVideoList(items);
+      }
+    });
+  }, []);
+
+  useEffect(() => {
+    if (draggableContainer.current) {
+      setDrake(
+        reactDragula([draggableContainer.current], {
+          moves: (_el, _container, handle) => {
+            if (!handle) {
+              return false;
+            }
+            return Boolean(handle.closest('.draggable'));
+          },
+        })
+      );
+    }
+  }, [draggableContainer.current]);
+
+  useEffect(() => {
+    const receiveMessage = (event: PopupObject) => {
+      const { getConfig, selectedVideosList } = event.data;
+      if (getConfig) {
+        event.source.postMessage(
+          {
+            message: 'Sending Config files',
+            config,
+            selectedVideos: videoList,
+          },
+          event.origin
+        );
+      } else if (selectedVideosList) {
+        saveExtensionData(selectedVideosList);
+      }
+    };
+    console.info('message :' + message);
+    window.addEventListener('message', receiveMessage, false);
+  });
+
+  const deleteVideo = (event: React.MouseEvent<HTMLElement>) => {
+    const videoId = event.currentTarget.getAttribute('data-id');
+
+    if (videoList) {
+      videoList.splice(
+        videoList.findIndex((index) => index.id === videoId),
+        1
+      );
+      console.log(videoList);
+
+      saveExtensionData([...videoList]);
+    } else {
+      saveExtensionData([]);
+    }
+  };
+
+  const saveExtensionData = (videos: VideoList[]) => {
+    let extensionData: VideoList[] | string[] = [];
+    if (config && !config.saveFullResponse) {
+      extensionData = videos.map((selected) => selected.id.toString());
+    } else {
+      extensionData = videos;
+    }
+    if (extension) {
+      extension.field.setData({ items: extensionData });
+      setVideoList(videos);
+    } else {
+      console.error('extension is not define');
+    }
+  };
+
+  const sonResponse = (error: object, res: ErrorMessage) => {
+    if (error) {
+      setMessage(res.message);
+    }
+  };
+
+  return (
+    <header className='App-header'>
+      <div className='wrapper'>
+        <div className='container'>
+          <div className='reference-loading' style={{ display: 'block' }}>
+            <div className='loading-flash'>
+              <div></div>
+              <div></div>
+              <div></div>
+              <div></div>
+            </div>
+          </div>
+          <div className='main'>
+            <div className='selected-item'>
+              <div className='row selected-list'>
+                <ul className='drag1' ref={draggableContainer}>
+                  {videoList &&
```

Comment:
Here too, I'm confused 
`videoList` always returns `true`, correct this condition